### PR TITLE
implemented model User, User Dao and User service, added HashUtil

### PR DESCRIPTION
**Pull request by Daniel1004K**

### CODE REVIEW

**Review by Tamara-Paskalova — APPROVED** [review]

File: src/main/java/mate/academy/spring/Main.java [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+    public static void main(String[] args) {
+        User user = new User();
+        user.setLogin("kuruch1004@gmail.com");
+        user.setPassword("password".toCharArray());
+
+        User user1 = new User();
+        user1.setLogin("kuruch@gmail.com");
+        user1.setPassword("password123".toCharArray());
+
+        User user2 = new User();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/Main.java:23]: I think we can name the variables better

File: src/main/java/mate/academy/spring/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+public class HashUtil {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/util/HashUtil.java:11]: let's make it a bean as well? and make methods non-static

File: src/main/java/mate/academy/spring/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    private static byte[] charsToBytes(char[] chars) {
+        final ByteBuffer byteBuffer = StandardCharsets.UTF_8.encode(CharBuffer.wrap(chars));
+        return Arrays.copyOf(byteBuffer.array(), byteBuffer.limit());
+    }
+
+    private static char[] bytesToChars(byte[] bytes) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/util/HashUtil.java:44]: do you use this method?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/util/impl/HashUtilImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import mate.academy.spring.util.HashUtil;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HashUtilImpl implements HashUtil {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/util/impl/HashUtilImpl.java:14]:
```suggestion
public class HashUtil {
```
we don't usually need interfaces for util logic

File: src/main/java/mate/academy/spring/util/HashUtil.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.spring.util;
+
+public interface HashUtil {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/util/HashUtil.java:3]: let's remove this interface

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**